### PR TITLE
databag: Adding new package

### PR DESCRIPTION
**Pull request by balzack**
Maintainer: Roland Osborne / @balzack
Compile tested: Raspberry Pi Zero W, Raspberry Pi 4b, OpenWrt v23.05.0
Run tested: Raspberry Pi Zero W, Raspberry Pi 4b, OpenWrt v23.05.0, tested federated messaging with browser and mobile app.

Description:
- Databag is a federated messaging service designed for self-hosting in consumer electronics. The service supports text, photos, video, and audio messages as well as audio and video calling, all with end-to-end encryption. The service provides browser and mobile apps (fdroid, android, and ios).
- Once the package is installed and running, the device admin should set UCI config for service_port and store_path.

Signed-off-by: Roland Osborne <roland.osborne@gmail.com>

### CODE REVIEW

**Comment by BKPepe** [comment]
> * Databag is a federated messaging service designed for self-hosting in consumer electronics

Is there any useful usage having this it on the router? I can imagine it on server, computer, etc. But on router?


> * The browser app is built with React, and I didn't think it was possible to include building the static web files as part of the build system.

It is possible, since I did similar thing with Turris WebApps, also created with React. I am not saying it was the best approach in that time, but its working.

This could be helpful for you as I implemented it with GitLab CI:
https://gitlab.nic.cz/turris/webapps/-/blob/35e83ff31946b3fa0b3e36c5441f37d00526aed7/.gitlab-ci.yml#L59

and then the package:
https://gitlab.nic.cz/turris/os/packages/-/blob/6d784bb5206c09e4102359a22eb1e0770223578c/web/turris-webapps/turris-webapps/Makefile

Also, if I remember correctly, I saw a few static websites for Go, but not sure if they were using React. But everything is possible. 

> * the device admin should set the service_port and store_path files under /etc/databag to the appropriate device configuration

Then the file should be marked as config file.

File: net/databag/Makefile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=databag
+PKG_VERSION:=1.1.27
+PKG_RELEASE:=1
+
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```
Comment (BKPepe) [inline net/databag/Makefile:7]: Source should be always packageName and version, not only version, otherwise dl folder gets messy.

**Comment by balzack** [comment]
@BKPepe 
- Is there any useful usage having this it on the router? I can imagine it on server, computer, etc. But on router?

For me routers, are the perfect device. As a privacy advocate my hope in a healthy positive internet lies with the decentralized web (not blockchain). I feel in order for these solutions to gain wide adoption, they need to be accessible to non-technical people. Cloud solutions are nice, but you get additional legal protections when you host your data from home. While OpenWrt is primarily used by the very technical, my hope is to one day be able to buy a router from a store that hosts all of my communication from the privacy of my own home.

I will look into the rest of your comments now. Thank you.

**Comment by balzack** [comment]
@BKPepe

I followed your example and now build the react webapp as a github action. Thank you for the suggestion!

It seems I have botched the commit formalities. Is it best if I close this PR and create a new one?

**Comment by BKPepe** [comment]
No, you dont need to create a new PR, just fix the existing one. We dont need to have 7 commits, but having just one is sufficient.

**Comment by balzack** [comment]
Sounds good, thanks; the commits have now been squashed.

**Comment by balzack** [comment]
@BKPepe 

Could I get some guidance as to why the mips24 test fails? I tried removing the mips specific flag, but that didn't help.

**Comment by balzack** [comment]
I think it's likely the test is failing because a Go dependency (https://gitlab.com/cznic/sqlite) doesn't support mips*_24kc. I've inquired there if there are any plans for support.

Are there any examples of go packages with sqlite? I'm not sure how to proceed.

**Comment by balzack** [comment]
I switched to a different sqlite driver for Go. I am still getting some build failures, but I think this is a better approach. The other driver seems to be a much larger lift in getting it to worth with mips*.

**Comment by BKPepe** [comment]
Was there something new since you are rebasing your branch so often? Looking at changes, I can not see any. Maybe I missed them.

**Comment by balzack** [comment]
The errors seem to be related to the docker build step:

ERROR: failed to solve: openwrt/rootfs:aarch64_generic-master: no match for platform in manifest sha256:4b3c818079b73f7cbe772e4f654e64975026caeaed694f41dfe5c9b9b1ff465d: not found

I think perhaps the docker image changed. This would not be related to the Databag project.

**Pull request closed without merging**